src/components/sidebar/sidebar.pw.tsx

```diff
@@ -0,0 +1,435 @@
+    await sidebarPreviewElement.press("Tab");
+    const focusedElement = page.locator("*:focus");
+    await focusedElement.press("Tab");
+    const firstInputElement = getDataElementByValue(page, "input").nth(0);
+
+    await expect(firstInputElement).toBeFocused();
+
+    await focusedElement.press("Tab");
+    const secondInputElement = getDataElementByValue(page, "input").nth(1);
+
+    await expect(secondInputElement).not.toBeFocused();
+
+    const openToastElement = getDataElementByValue(page, "open-toast");
+
+    await expect(openToastElement).toBeFocused();
+  });
+
+  test("should return focus to the Toast within component after non-focusable content has been selected", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<SidebarComponentFocusable />);
+
+    const toastElement = getComponent(page, "toast");
+
+    await expect(toastElement).not.toBeVisible();
+
+    const openToastElement = getDataElementByValue(page, "open-toast");
+    await openToastElement.click();
+
+    await expect(toastElement).toBeVisible();
+
+    await page.keyboard.press("Tab");
+    const closeIconButtonElement = closeIconButton(page).nth(1);
+
+    await expect(closeIconButtonElement).toBeFocused();
+  });
+
+  test("should call onCancel callback when a click event is triggered", async ({
+    mount,
+    page,
+  }) => {
+    let callbackCount = 0;
+    await mount(
+      <SidebarComponent
+        onCancel={() => {
+          callbackCount += 1;
+        }}
+      />
+    );
+
+    const closeIconButtonElement = closeIconButton(page);
+    await closeIconButtonElement.click();
+
+    await expect(callbackCount).toEqual(1);
+  });
+
+  test("should ensure the component is rendered on top of any other modals when the topModalOverride prop is true", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<TopModalOverride />);
+
+    const sidebarPreviewElement = sidebarPreview(page);
+
+    await expect(sidebarPreviewElement).toBeVisible();
+
+    const bodyElement = page.locator("body");
+    await bodyElement.press("Tab");
+    const sidebarPreviewCloseIconButton = sidebarPreview(page).locator(
+      CLOSE_ICON_BUTTON
+    );
+
+    await expect(sidebarPreviewCloseIconButton).toBeFocused();
+
+    const focusedElement = page.locator("*:focus");
+    await focusedElement.press("Enter");
+
+    await expect(sidebarPreviewElement).not.toBeVisible();
+
+    await bodyElement.press("Tab");
+    const DialogCloseIconButton = getComponent(page, "dialog").locator(
+      CLOSE_ICON_BUTTON
+    );
+
+    await expect(DialogCloseIconButton).toBeFocused();
+  });
+});
+
+test.describe("Accessibility tests for Sidebar component", () => {
+  test("should check accessibility for SidebarComponent example", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<SidebarComponent />);
+
+    await checkAccessibility(page);
+  });
+
+  (["left", "right"] as SidebarProps["position"][]).forEach((boolVal) => {
```

Comment:
**comment**: quite apart from my usual complainat about `boolVal` as a variable name, this isn't even boolean! Let's just call it `position` or something similar.